### ISSUE TITLE
add CUE API support for non-Go languages

### ISSUE THREAD

**Issue by 4ad** [cue-lang/cue#3104]
Currently CUE only has a Go API. We're planning to support more languages.

This is an umbrella issue for discussing adding CUE support to more languages, as an API.

The CUE API is exposed as a C-level API in [libcue](https://github.com/cue-lang/libcue). On top of that, we build language-specific APIs. Currently we are working on [cue-api-java](https://github.com/cue-lang/cue-api-java) for Java and [cue-api-python](https://github.com/cue-lang/cue-api-python) for Python. Rust support also planned.

We have opened some project specific umbrella issues:
* libcue: #3103 
* cue-api-java: #3101 
* cue-api-python: #3100 
* cue-api-rust: #3102

**Comment by NoamTD** [comment]
@4ad are there any plans to add support for C#?

**Comment by 4ad** [comment]
We haven't yet planned what languages we will support after Java, Python and Rust, but we wish to support as many languages as possible, so we're looking for feedback from people on what to target next.

Part of making a good API is being idiomatic for that specific language, if you or somebody else is an expert in C#, we also welcome help in designing such an API.

Additionally, if the community wants to add support for a specific language, please get in touch with us so we could help you and make your job easier.

**Comment by mvdan** [comment]
There is also a GitHub project to track the ongoing and planned work, which I imagine Aram meant to link to above: https://github.com/orgs/cue-lang/projects/24

**Comment by myitcv** [comment]
Quick heads-up for folks finding this thread via other routes. Per our [Community Call updates](https://github.com/cue-lang/cue/discussions/categories/community-calls), the CUE + WASM effort is on pause. In particular, the work to use WASM as a cross-language bridge for embedding the evaluator is currently on hold. 

Some context on that decision:

 • Ongoing flux in the WASM/WASI/WIT/component model standards landscape.  
 • The approach depends on [cue-lang/cue\#2007](https://github.com/cue-lang/cue/issues/2007), which we consider a critical requirement.  
 • That strand was also gated on eval v3 landing (which it now has).

As ever, we’re happy to re-evaluate priorities—if there’s renewed interest or concrete use cases here, please shout.
